src/ChainSafe.Gaming.Reown/ReownProvider.cs

```diff
@@ -463,7 +467,12 @@ public override async Task<T> Request<T>(string method, params object[] paramete
                 handler => SignClient.CoreClient.Relayer.Publisher.OnPublishedMessage += handler,
                 handler => SignClient.CoreClient.Relayer.Publisher.OnPublishedMessage -= handler);
 
-            return await ReownRequest<T>(sessionTopic, method, parameters);
+            // For whatever reason, android and iOS are forcefully killing our thread where this is being run on
+            // So we are ensuring that the request survives the thread kill by running it on the main thread
+            return
+                osMediator.Platform == Platform.Android || osMediator.Platform == Platform.IOS ?
+                    await Task.Run(() => ReownRequest<T>(sessionTopic, method, parameters))
```

Comment:
It looks like we're doing the opposite to what you stated in the comment, aren't we? If it's Android or iOS we run it on the worker thread with `Task.Run()`.

Reply:
Ah I have ommitted the mainthreaddispatcher completley. It works like this 🤷‍♂️
Maybe you can try and add it but I really think it will not work since if its on the main thread it will be stopped as soon as you go out of the app.

Reply:
If it works, it works. Just need to update the comment.